Makefile

```diff
@@ -72,4 +72,7 @@ e2e:
 e2e-prod:
 	FOCUS="\[Real\]" ./hack/e2e.sh
 
+e2e-keyrotation:
```

Comment:
This should be a target which will run all long running tests.  We'd run this via a prow periodic.

Reply:
are we saying we want to test this with each e2e run?

Reply:
We need to be able to run these long running tests individually. At that point, I am not sure what a single job that runs all long-running tests gives us since each test is going to create/update its own cluster.

Reply:
> are we saying we want to test this with each e2e run?

No, these tests should run periodically and we should also be able to run them individually and optionally on PRs.

Reply:
I think the idea here is to condense all of the make e2e targets into a single `make e2e` and then we tag the various e2e specs as follows:

`[EveryPR]` - These tests should be run on every PR (duration-based tag)
`[LongRunning]` - These tests are long running and should be run periodically (duration-based tag)
`[Real]` - These tests should be run against the real RP (endpoint-based tag)
`[Fake]` - These tests should be run against the fake RP until the admin API is ready (endpoint-based tag)
`[RP]` - These tests could be run against either the fake or real RP after the admin API is ready (endpoint-based tag)

The developer of a spec is free to add more tags however those 5 above should be the supported ones. For example:

* Key rotation test (before admin api): `[KeyRotation] [Fake] [LongRunning]`
* Key rotation test (after admin api): `[KeyRotation] [RP] [LongRunning]`
* Scale up/down test (before admin api): `[ScaleUpDown] [Fake] [Real] [LongRunning] [EveryPR]`
* Scale up/down test (after admin api): `[ScaleUpDown] [RP] [LongRunning] [EveryPR]`

Then we create prow jobs using using focus combinations to filter through eligible tests:

* Run all long running tests in one job: `FOCUS=[LongRunning] make e2e`
  * Run the long running key rotation test in one job: `FOCUS=[KeyRotation] make e2e`
  * Run the long running scale up/down test in one job: `FOCUS=[ScaleUpDown] make e2e`
* Run all tests against the real RP: `FOCUS=Real make e2e`
* Run the set of tests before every PR in one job: `FOCUS=EveryPR make e2e`
* Run all tests against the fake RP (until admin API is ready): `FOCUS=Fake make e2e`
* Run all tests against any RP (after admin API is ready): `FOCUS=RP make e2e`

Reply:
Agreed. And all this sound like small readme in docs/tests.md with convention or smth like that. as I think this will come up again :)